sdk/resourcemanager/Azure.ResourceManager.Core/src/Generated/PreDefinedTagContainer.cs

```diff
@@ -0,0 +1,217 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+
+namespace Azure.ResourceManager.Core
+{
+    /// <summary>
+    /// A class representing collection of Tag and its operations.
+    /// </summary>
+    public class PreDefinedTagContainer : ContainerBase
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PreDefinedTagContainer"/> class for mocking.
+        /// </summary>
+        protected PreDefinedTagContainer()
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PreDefinedTagContainer"/> class.
+        /// </summary>
+        /// <param name="clientContext">Current client context. </param>
+        /// <param name="subscription">The parent subscription. </param>
+        internal PreDefinedTagContainer(ClientContext clientContext, SubscriptionResourceIdentifier subscription)
+            : base(clientContext, new SubscriptionResourceIdentifier(subscription))
+        {
+            RestClient = new TagsRestOperations(Diagnostics, Pipeline, subscription.SubscriptionId, BaseUri);
+        }
+
+        /// <summary>
+        /// Gets the valid resource type associated with the container.
+        /// </summary>
+        protected override ResourceType ValidResourceType => SubscriptionOperations.ResourceType;
+
+        /// <summary>
+        /// Gets the operations that can be performed on the container.
+        /// </summary>
+        private TagsRestOperations RestClient;
+
+        /// <summary> This operation allows adding a name to the list of predefined tag names for the given subscription. A tag name can have a maximum of 512 characters and is case-insensitive. Tag names cannot have the following prefixes which are reserved for Azure use: &apos;microsoft&apos;, &apos;azure&apos;, &apos;windows&apos;. </summary>
+        /// <param name="tagName"> The name of the tag to create. </param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        public virtual async Task<Response<PreDefinedTag>> CreateOrUpdateAsync(string tagName, CancellationToken cancellationToken = default)
+        {
+            using var scope = Diagnostics.CreateScope("PreDefinedTagContainer.CreateOrUpdate");
+            scope.Start();
+            try
+            {
+                var response =  await StartCreateOrUpdateAsync(tagName, cancellationToken).ConfigureAwait(false);
+                return Response.FromValue(response.Value, response.GetRawResponse());
```

Comment:
need to call WaitForCompletionAsync here since Value may not be filled in

Reply:
Done